hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -548,13 +548,15 @@ private class Emit(
         val vab = new StagedArrayBuilder(atyp.elementType, mb, 16)
         val sorter = new ArraySorter(er, vab)
 
-        val (array, compare, distinct) = (x: @unchecked) match {
-          case ArraySort(a, l, r, comp) => (a, Subst(comp, BindingEnv(Env[IR](l -> In(0, eltType), r -> In(1, eltType)))), Code._empty[Unit])
+        val (array, compare, distinct, fBindings: Array[String]) = (x: @unchecked) match {
```

Comment:
Favor over Map, because the indices that were provided (into newMethod args), required knowledge by the caller, and an array is a clearer data structure for ordered collections (order here: left binding name, right binding name)

Reply:
oh, yeah, totally.